### PR TITLE
PLGN 82 and 87 new

### PR DESCRIPTION
**Pull request by pritesh-lr**
The purpose of this PR is to fix below issues:

1. PLGN-82: Magento Cardknox plugin compatible with php 8.2 and magento v 2.4.6
2. PLGN-87: Security updates
3. Update iFields latest version 2.15.2302.0801
4. Fix-Run automated PHPunit test

### CODE REVIEW

File: .github/workflows/run-magento-tests.yml [hunk 1]
```diff
@@ -5,7 +5,10 @@ on:
     branches-ignore: 
       - 'master'
       - 'rc/*/*/*'
-      
+  pull_request:
```
Comment (mstein-ck) [inline .github/workflows/run-magento-tests.yml:8]: Why is this needed?

File: .github/workflows/run-magento-tests.yml [hunk 2]
```diff
@@ -65,4 +68,5 @@ jobs:
     - name: Run Unit Tests
       run: |
         cd magento
+        sudo sed -i "/<string>allure.*/c\<directory>allure/allure.config.php</directory>" dev/tests/unit/phpunit.xml.dist
```
Comment (mstein-ck) [inline .github/workflows/run-magento-tests.yml:71]: What is this change for?

File: .github/workflows/run-magento-tests.yml (hunk 1 above)
Comment (pritesh-lr) [reply .github/workflows/run-magento-tests.yml:8]: Actually, **on push**, it only works for the main Carknox repo, and the branch is not master; when I use the forked repo, so GitHub workflow works for `pritesh-lr:<other_branch>`, but When I create a Pull request, it still fails GitHub workflow, 
By adding the above code, it will work for both times **on push** if a branch is not master and **on pull request open.**

File: .github/workflows/run-magento-tests.yml (hunk 2 above)
Comment (pritesh-lr) [reply .github/workflows/run-magento-tests.yml:71]: In the latest magento 2.4.6, When we run the PHP unit test, it throws an error: `Error: Config file allure/allure.config.php doesn't exist`,
By adding this line, it fixes the error and run the PHPUnit test successfully.
